### PR TITLE
feat: add get players rpc, start game logic and grpc exception interceptor

### PR DESCRIPTION
**Pull request by Kuruyia**
This adds a new route to get all the players of a game.

This also adds the route and logic to start the game, which ensures that there are at least 8 players registered.

A global gRPC server interceptor has been added to the dedicated server to catch thrown Kotlin exceptions and translate them to gRPC status codes.

### CODE REVIEW

File: common/src/main/kotlin/fr/agrouagrou/common/GameManager.kt [hunk 1]
```diff
@@ -2,8 +2,19 @@ package fr.agrouagrou.common
...
 
 class GameManager {
     val gameState = GameState()
-    val playerManager = PlayerManager()
+    val playerManager = PlayerManager(gameState)
+
+    fun startGame() {
+        if (gameState.status.value != GameStateStatus.LOOKING_FOR_PLAYERS) {
+            throw IllegalStateException("Game is already started")
+        } else if (playerManager.players.size < 8) {
```
Comment (mmoreiradj) [inline common/src/main/kotlin/fr/agrouagrou/common/GameManager.kt]: Can you parametrize the min size ?

File: common/src/main/kotlin/fr/agrouagrou/common/player/PlayerRegistry.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package fr.agrouagrou.common.player
+
+import java.util.*
+
+class PlayerRegistry {
+    private val mutablePlayers = mutableMapOf<UUID, Player>()
+    val players = mutablePlayers as Map<UUID, Player>
+
+    fun register(username: String): Player {
+        val player = Player(username)
```
Comment (mmoreiradj) [inline common/src/main/kotlin/fr/agrouagrou/common/player/PlayerRegistry.kt]: How does the host know which player is which if you use a generated ID and not check for duplicate usernames ?

**Review by mmoreiradj — CHANGES_REQUESTED** [review]

File: common/src/main/kotlin/fr/agrouagrou/common/player/PlayerRegistry.kt (hunk 2 above)
Comment (Kuruyia) [reply common/src/main/kotlin/fr/agrouagrou/common/player/PlayerRegistry.kt]: Good point, I thought I had the ID in the reply, but apparently I removed it at some point. I'll add that back to the reply.

Comment (Kuruyia) [reply common/src/main/kotlin/fr/agrouagrou/common/player/PlayerRegistry.kt]: Just checked, but the player ID is in fact already sent back to the client.

**Review by mmoreiradj — APPROVED** [review]
lgtmm

**Pull request merged**